encoder_test.go

```diff
@@ -83,6 +83,7 @@ func TestTextDecoder(t *testing.T) {
 		{"Enum16('one'=1)", "'one'", "one"},
 		{"Array(UInt32)", "[1,2]", []uint32{1, 2}},
 		{"Array(UInt32)", "[]", []uint32{}},
+		{"Array(String)", "['one','one\\'']", []string{"one", "one'"}},
```

Comment:
this test passes on current HEAD without patch for method `func (d *textDecoder) Decode(t string, value []byte) (driver.Value, error)`